sycl/cmake/modules/AddSYCLUnitTest.cmake

```diff
@@ -6,7 +6,7 @@ macro(add_sycl_unittest test_dirname link_variant)
   # Enable exception handling for these unit tests
   set(LLVM_REQUIRES_EH 1)
 
-  if (MSVC AND CMAKE_BUILD_TYPE MATCHES "Debug")
+  if (MSVC AND build_type_lower MATCHES "debug")
```

Comment:
Actually I thought about moving duplicated code within `sycl/cmake/modules/AddSYCLUnitTest.cmake` only.
How about adding a new macro for configurations common for both `add_sycl_unittest` and `add_sycl_unittest_with_device`?
E.g. 
```cmake 
macro(add_sycl_unittest_common CONFIG_SUFFIX)
  # Enable exception handling for these unit tests
  set(LLVM_REQUIRES_EH 1)
  string(TOLOWER "${CMAKE_BUILD_TYPE}" build_type_lower)
  if (MSVC AND build_type_lower MATCHES "debug")
    set (CONFIG_SUFFIX "d")
  endif
endmacro()

macro(add_sycl_unittest_with_device test_dirname link_variant)
  add_sycl_unittest_common(CONFIG_SUFFIX)
  set(sycl_obj_target "sycl${CONFIG_SUFFIX}_object")
  set(sycl_so_target "sycl${CONFIG_SUFFIX}")
  set(XPTI_LIB xpti${CONFIG_SUFFIX})
...
```

Reply:
`add_sycl_unittest_with_device` was dismissed, so I've reverted some changes.